File: erezervisi_api/eRezervisi/eRezervisi.Common.Dtos/User/UserGetDto.cs
using eRezervisi.Common.Dtos.Role;
using eRezervisi.Common.Dtos.UserSettings;

namespace eRezervisi.Common.Dtos.User
{
    public class UserGetDto
    {
        public long Id { get; set; }
        public string FirstName { get; set; } = null!;
        public string LastName { get; set; } = null!;
        public string Phone { get; set; } = null!;
        public string Address { get; set; } = null!;
        public string Email { get; set; } = null!;
        public string? Image { get; set; }
        public long RoleId { get; set; }
        public RoleGetDto Role { get; set; } = null!;
        public string Username { get; set; } = null!;
        public UserSettingsGetDto UserSettings { get; set; } = null!;
    }
}

File: erezervisi_desktop/erezervisi_desktop/lib/models/responses/user/user_get_dto.dart
import 'package:erezervisi_desktop/models/responses/user/user_settings_get_dto.dart';

class UserGetDto {
  late num id;
  late String firstName;
  late String lastName;
  late String phone;
  late String address;
  late String email;
  late String? image;
  late String username;
  late UserSettingsGetDto settings;

  UserGetDto(
      {required this.id,
      required this.firstName,
      required this.lastName,
      required this.phone,
      required this.address,
      required this.email,
      required this.username,
      this.image,
      required this.settings});

  factory UserGetDto.fromJson(Map<String, dynamic> json) {
    return UserGetDto(
        id: json['id'] as num,
        firstName: json['firstName'] as String,
        lastName: json['lastName'] as String,
        phone: json['phone'] as String,
        address: json['address'] as String,
        email: json['email'] as String,
        username: json['username'] as String,
        image: json['image'],
        settings: UserSettingsGetDto.fromJson(json['userSettings']));
  }
}
